lib/omniauth/strategies/openid_connect.rb

```diff
@@ -112,15 +133,15 @@ def callback_phase
         raise CallbackError, error: params['error'], reason: error_description, uri: params['error_uri'] if error
         raise CallbackError, error: :csrf_detected, reason: "Invalid 'state' parameter" if invalid_state
 
-        return unless valid_response_type?
+        return if configured_response_type == 'code' && !valid_response_type?
 
         options.issuer = issuer if options.issuer.nil? || options.issuer.empty?
 
-        verify_id_token!(params['id_token']) if configured_response_type == 'id_token'
+        verify_id_token!(params['id_token']) if configured_response_type == 'id_token token'
```

Comment:
I think there should be some method for `implicit_flow?`, `client_credentials_flow?`, etc. There are many possibilities for `response_type`.

Reply:
Ah, I see.
I want to organize it with the patch part 2 to push next.